### PR TITLE
Simplify online game status

### PR DESCRIPTION
**Pull request by zorgjbe**
As per the nice description from [Discord](https://discord.com/channels/554377975714414605/554378725916147722/935977890850410557), the online games' status can get messed up by inconsistent values received from players.

This PR adds some typing information for the online game parts, then proceeds to add helper functions and remove the duplicated local & server-supplied variables, making sure they only take or return `""` or `"Running"` as valid values.

There's some cleanup and bugfixing done to the MagicBattle click-handler as well, as it would toggle the game on and off on startup since the locking was now done on the same variable.

### CODE REVIEW

File: BondageClub/Screens/Online/GameMagicBattle/GameMagicBattle.js [hunk 1]
```diff
@@ -11,6 +10,33 @@ var GameMagicBattleFocusCharacter = null;
...
+ */
+function GameMagicBattleGetStatus() {
+	if (Player.Game && Player.Game.MagicBattle && ["", "Running"].includes(Player.Game.MagicBattle.Status))
+		return Player.Game.MagicBattle.Status;
+	return "";
+}
+
+/**
+ * Set the current state of LARP.
+ * @param {string} s
```
Comment (Sidiousious) [inline BondageClub/Screens/Online/GameMagicBattle/GameMagicBattle.js:25]:
```suggestion
 * @param {"" | "Running"} s
```

Comment (Sidiousious) [reply BondageClub/Screens/Online/GameMagicBattle/GameMagicBattle.js:25]: consider union type with string literals of the valid options

File: BondageClub/Screens/Online/GameLARP/GameLARP.js [hunk 2]
```diff
@@ -51,13 +50,40 @@ var GameLARPTurnTimerDelay = GameLARPTimerDelay[0];
...
+ */
+function GameLARPGetStatus() {
+	if (Player.Game && Player.Game.LARP && ["", "Running"].includes(Player.Game.LARP.Status))
+		return Player.Game.LARP.Status;
+	return "";
+}
+
+/**
+ * Set the current state of LARP.
+ * @param {string} s
```
Comment (Sidiousious) [inline BondageClub/Screens/Online/GameLARP/GameLARP.js:65]:
```suggestion
 * @param {"" | "Running"} s
```

**Comment by zorgjbe** [comment]
Thanks for the review @Sidiousious. I've switched the helpers to take and return the typedef itself.

**Comment by Ben987** [comment]
Thanks!  And how can I recreate the bug that this fixes in the current version?  What are the steps?

**Comment by zorgjbe** [comment]
See the Discord link in the OP, which have a full explanation and steps, either programmatically or "what could be going on", which I'll reproduce here :
```
Make a LARP room
Get a player who has not played a game of LARP to join
Make sure the new player has a class and team selected
Make the new player admin
The new player is now hard glitched, and you are soft glitched
```
Console-way: `GameLARPStatus = undefined; Player.Game.LARP.Status = undefined; ServerAccountUpdate.QueueData({ Game: Player.Game }, true);`

All of this might (and this is speculation, so grain of salt) be because for new players that haven't been through the LARP room, the server's `Player.Game` object doesn't exist.

**Comment by Ben987** [comment]
Great!  I will test that bug later today or tomorrow to see if I can recreate it and to see if your solution fixes it.  Cheers :)

**Comment by Ben987** [comment]
I was able to recreate the bug, lets see if that fixes it.  Cheers!

**Comment by Ben987** [comment]
Sadly, the fix created a new bug where it's not possible to start a LARP game anymore.  Thanks for reporting the bug though, I'll try to find a simpler way to fix it.

**Pull request merged**